### ISSUE TITLE
How do you tell who the extracted memories (facts, preferences, etc.) belong to?

### ISSUE THREAD

**Issue by littttttlebird** [mem0ai/mem0#1528]
For example, in a companion app, the user (alice) says to the assistant (bob)："Yesterday my good friend jack said that he didn't like playing football, but his father enrolled him in a football training class. It bothered him a lot."

For this sentence, the memory extracted might be: "don't like playing football "," attended a football training class " or something like this。
However, these memories, which should belong to the third party "jack", will be remembered by alice on the spot in this project.
How to solve this problem?

**Comment by Dev-Khant** [comment]
Hey @littttttlebird Great question, I tried the same prompt with Mem0 platform as it solves the exact problem you talked about. It will remember that we are talking about `jack` here. So here you can keep `user_id` as alice.

Here are the memories created for it:

<img width="596" alt="Screenshot 2024-08-02 at 7 23 36 PM" src="https://github.com/user-attachments/assets/521f2570-8960-488f-bcb9-8a15179a716e">

**Comment by littttttlebird** [comment]
Awesome, I asked this question based on this line in the prompt:
“- Don't remember the user/agent details provided. Only remember the facts, preferences, and memories."
It looks like the third party information in the conversation is interpreted as fact, not user detail.

**Comment by Dev-Khant** [comment]
> Awesome, I asked this question based on this line in the prompt: “- Don't remember the user/agent details provided. Only remember the facts, preferences, and memories." It looks like the third party information in the conversation is interpreted as fact, not user detail.

Yes you are correct but we have much better prompt for platform :)

**Comment by parshvadaftari** [comment]
Closing as resoleved.
